clients/js/src/Collection.ts

```diff
@@ -438,76 +438,76 @@ export class Collection {
+     * @param {Embedding | Embeddings} [params.queryEmbeddings] - Optional query embeddings to use for the search.
+     * @param {PositiveInteger} [params.nResults] - Optional number of results to return (default is 10).
      * @param {Where} [params.where] - Optional query condition to filter results based on metadata values.
-     * @param {string | string[]} [params.query_texts] - Optional query text(s) to search for in the collection.
-     * @param {WhereDocument} [params.where_document] - Optional query condition to filter results based on document content.
+     * @param {string | string[]} [params.queryTexts] - Optional query text(s) to search for in the collection.
+     * @param {WhereDocument} [params.whereDocument] - Optional query condition to filter results based on document content.
      * @param {IncludeEnum[]} [params.include] - Optional array of fields to include in the result, such as "metadata" and "document".
      *
      * @returns {Promise<QueryResponse>} A promise that resolves to the query results.
      * @throws {Error} If there is an issue executing the query.
      * @example
      * // Query the collection using embeddings
      * const results = await collection.query({
-     *   query_embeddings: [[0.1, 0.2, ...], ...],
-     *   n_results: 10,
+     *   queryEmbeddings: [[0.1, 0.2, ...], ...],
+     *   nResults: 10,
      *   where: {"name": {"$eq": "John Doe"}},
      *   include: ["metadata", "document"]
      * });
      * @example
      * ```js
      * // Query the collection using query text
      * const results = await collection.query({
-     *   query_texts: "some text",
-     *   n_results: 10,
+     *   queryTexts: "some text",
+     *   nResults: 10,
      *   where: {"name": {"$eq": "John Doe"}},
      *   include: ["metadata", "document"]
      * });
      * ```
      *
      */
     public async query({
-        query_embeddings,
-        n_results,
+        queryEmbeddings,
+        nResults,
         where,
-        query_texts,
-        where_document,
+        queryTexts,
+        whereDocument,
         include,
     }: {
-        query_embeddings?: Embedding | Embeddings,
-        n_results?: PositiveInteger,
+        queryEmbeddings?: Embedding | Embeddings,
+        nResults?: PositiveInteger,
         where?: Where,
-        query_texts?: string | string[], // TODO: should be named query_texts to match python API
-        where_document?: WhereDocument, // {"$contains":"search_string"}
+        queryTexts?: string | string[],
+        whereDocument?: WhereDocument, // {"$contains":"search_string"}
         include?: IncludeEnum[] // ["metadata", "document"]
     }): Promise<QueryResponse> {
-        if (n_results === undefined) n_results = 10
-        if (query_embeddings === undefined && query_texts === undefined) {
+        if (nResults === undefined) nResults = 10
+        if (queryEmbeddings === undefined && queryTexts === undefined) {
```

Comment:
down the road you'll want to also verify this statically instead of just having all params be optional.

TS type system is very powerful, and you can do a lot of this type of logic at the type level – while also having the validation logic client-side (and i'm sure you're doing similar validation logic server-side as well).

not a big deal for now, but it's these small DX things which make a difference